vendor/github.com/dop251/goja/README.md

```diff
@@ -212,34 +212,8 @@ There are two standard mappers: [TagFieldNameMapper](https://godoc.org/github.co
 Native Constructors
 -------------------
 
-In order to implement a constructor function in Go:
-```go
-func MyObject(call goja.ConstructorCall) *Object {
-    // call.This contains the newly created object as per http://www.ecma-international.org/ecma-262/5.1/index.html#sec-13.2.2
-    // call.Arguments contain arguments passed to the function
-
-    call.This.Set("method", method)
-
-    //...
-
-    // If return value is a non-nil *Object, it will be used instead of call.This
-    // This way it is possible to return a Go struct or a map converted
-    // into goja.Value using runtime.ToValue(), however in this case
-    // instanceof will not work as expected.
-    return nil
-}
-
-runtime.Set("MyObject", MyObject)
-
-```
-
-Then it can be used in JS as follows:
-
-```js
-var o = new MyObject(arg);
-var o1 = MyObject(arg); // same thing
-o instanceof MyObject && o1 instanceof MyObject; // true
-```
+In order to implement a constructor function in Go use `func (goja.ConstructorCall) *goja.Object`.
```

Comment:
Hmm how does this affect our `XConstructor` magic?

Reply:
In no way ... as [we are not using it](https://github.com/loadimpact/k6/blob/master/js/common/bridge.go#L258-L263)

But also this seems as only a documentation chang

Reply:
But apparently there's a native way to do this in goja which we don't use. Should we open an issue to investigate this, given that this is the code above our code to do the same :sweat_smile: https://github.com/loadimpact/k6/blob/ccffbcaaddd7f2d472cdced0da847fde2140ef88/js/common/bridge.go#L257

Reply:
I don't think I will come to this until we decide to rewrite the bridge :D ... so if you think you will come to this refactoring work before that ... or think someone else will do it - open an issue ;)